app/controllers/customers_controller.rb

```diff
@@ -11,19 +11,28 @@ class CustomersController < Sellers::BaseController
   layout "inertia", only: [:index]
 
   def index
-    product = Link.fetch(params[:link_id]) if params[:link_id].present?
-    sales = fetch_sales(products: [product].compact)
-    customers_presenter = CustomersPresenter.new(
-      pundit_user:,
-      product:,
-      customers: load_sales(sales),
-      pagination: { page: 1, pages: (sales.results.total / CUSTOMERS_PER_PAGE.to_f).ceil, next: nil },
-      count: sales.results.total
-    )
-    create_user_event("customers_view")
+    if !request.inertia_partial?
+      product = Link.fetch(params[:link_id]) if params[:link_id].present?
+      sales = fetch_sales(products: [product].compact)
+      customers_presenter = CustomersPresenter.new(
+        pundit_user:,
+        product:,
+        customers: load_sales(sales),
+        pagination: { page: 1, pages: (sales.results.total / CUSTOMERS_PER_PAGE.to_f).ceil, next: nil },
+        count: sales.results.total
+      )
+      create_user_event("customers_view")
+    end
+
+    purchase = current_seller.sales.find_by_external_id!(params[:purchase_id]) if params[:purchase_id].present?
 
-    render inertia: "Customers/Index",
-           props: { customers_presenter: customers_presenter.customers_props }
+    render inertia: "Customers/Index", props: {
+      customers_presenter: -> { customers_presenter.customers_props },
+      customer_emails: InertiaRails.optional { CustomerPresenter.new(purchase:).customer_emails },
+      missed_posts: InertiaRails.optional { CustomerPresenter.new(purchase:).missed_posts(workflow_id: params[:workflow_id]) },
+      workflows: InertiaRails.optional { WorkflowsPresenter.new(seller: current_seller).workflow_options_by_purchase_props(purchase) },
+      product_purchases: (purchase&.is_bundle_purchase? ? InertiaRails.optional { purchase.product_purchases.map { CustomerPresenter.new(purchase: _1).customer(pundit_user:) } } : nil),
```

Comment:
Ideally this change should be in a new PR but I had to make this a part of props as well to simplify state reconciliation on frontend. 

Without this I'd have had to manage two network calls to sync component state.

I have moved the specs for the removed controller action accordingly, lmk if this is undesired